### PR TITLE
[browser][MT] use portable timer

### PR DESCRIPTION
**Pull request by pavelsavara**
`TimerQueue` ST implementation relies on `setTimeout` (which deadlocks in MT)
- we are moving `TimerQueue` from MT+ST to ST only. 
- After this change, MT will use portable timer (same as any Unix), which will create dedicated thread.

### CODE REVIEW

**Comment by kg** [comment]
This looks like timers will only ever run on the main thread, is that a problem? I don't recall whether timers work that way on other platforms

**Review by kg — APPROVED** [review]

File: src/mono/System.Private.CoreLib/System.Private.CoreLib.csproj [hunk 1]
```diff
@@ -288,6 +285,7 @@
       <Compile Include="$(BclSourcesRoot)\System\Threading\ThreadPoolBoundHandle.Browser.Mono.cs" />
       <Compile Include="$(BclSourcesRoot)\System\Threading\PreAllocatedOverlapped.Browser.Mono.cs" />
       <Compile Include="$(BclSourcesRoot)\System\Threading\ThreadPool.Browser.Mono.cs" />
+      <Compile Include="$(BclSourcesRoot)\System\Threading\TimerQueue.Browser.Mono.cs" />
```
Comment (ilonatommy) [inline src/mono/System.Private.CoreLib/System.Private.CoreLib.csproj:288]: Why are we adding it when `FeatureWasmThreads=false`? Aren't we making changes for MT-only and non-MT should stay as it was?

**Review by ilonatommy — APPROVED** [review]

**Comment by pavelsavara** [comment]
> This looks like timers will only ever run on the main thread, is that a problem? I don't recall whether timers work that way on other platforms

We discussed this elsewhere, portable timer in MT will create dedicated thread. ST will stay using `setTimeout`

**Pull request merged**